src/Controller/Component/AuthenticationComponent.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\Controller\Component;
+
+use Cake\Controller\Component;
+use Cake\Event\EventDispatcherTrait;
+use Cake\Utility\Hash;
+use Exception;
+
+class AuthenticationComponent extends Component
+{
+
+    use EventDispatcherTrait;
+
+    /**
+     * Authentication service instance.
+     *
+     * @var \Authentication\AuthenticationService
+     */
+    protected $_authentication;
+
+    /**
+     * Initialize component.
+     *
+     * @param array $config The config data.
+     * @return void
+     */
+    public function initialize(array $config)
+    {
+        $controller = $this->_registry->getController();
+        $this->_authentication = $controller->request->getAttribute('authentication');
+
+        if ($this->_authentication === null) {
```

Comment:
We should check for the instance not just if it's not null. This also brings up the question if we don't want to implement an interface for the service so people can change the service without breaking stuff.